tests/unit/scanners/test.javascript.js

```diff
@@ -103,6 +103,53 @@ describe('JavaScript Scanner', () => {
     expect(linterMessages.length).toEqual(0);
   });
 
+  it('should support optional chaining', async () => {
+    const code = 'const dogName = adventurer.dog?.name;';
+
+    const jsScanner = new JavaScriptScanner(code, 'code.js');
+
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
+  });
+
+  it('should support nullish coalescing operator', async () => {
+    const code = 'const baz = 0 ?? 42;';
+
+    const jsScanner = new JavaScriptScanner(code, 'code.js');
+
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
+  });
+
+  // eslint-disable-next-line jest/no-disabled-tests
+  it.skip('should support public class fields', async () => {
+    const code = 'class MyClass { a = 1; }';
+
+    const jsScanner = new JavaScriptScanner(code, 'code.js');
+
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
+  });
+
+  it('should support BigInt short-hand notation', async () => {
+    const code = 'const bigInt = 2166136261n;';
+
+    const jsScanner = new JavaScriptScanner(code, 'code.js');
+
+    const { linterMessages } = await jsScanner.scan();
+    expect(linterMessages.length).toEqual(0);
+  });
+
+  // eslint-disable-next-line jest/no-disabled-tests
+  it.skip('should support dynamic imports', async () => {
```

Comment:
I thought this would work but it does not.. The promise version seems to work but only because it is a promise and not because it is dynamic import.

Reply:
> ```
>    const code = "let { hi, bye } = await import('./say.js');";
>```
Is this a top-level `await`? That syntax is not supported yet by the JS language. It's still a proposal: https://github.com/tc39/proposal-top-level-await

To use `await`, you need to wrap it in an async function.

Reply:
I know that but my understanding is that it should work, according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import

Reply:
I don't see anything on that page that claims that `await` in the top-level scope to work. In fact, the complete examples over there are also wrapped in an async IIFE.

Reply:
mm, right. I'll update the test case then.